packages/@aws-cdk/aws-codepipeline-actions/lib/stepfunctions/invoke-action.ts

```diff
@@ -0,0 +1,159 @@
+   *
+   * @default the Action will not have any outputs
+   */
+  readonly output?: codepipeline.Artifact;
+
+  /**
+   * The state machine to invoke.
+   */
+  readonly stateMachine: stepfunction.IStateMachine;
+
+  /**
+   * Represents the input to the StateMachine.
+   * This includes input artifact, input type and the statemachine input.
+   *
+   * @default - none
+   */
+  readonly stateMachineInput?: StateMachineInput;
+
+  /**
+   * Prefix (optional)
+   *
+   * By default, the action execution ID is used as the state machine execution name.
+   * If a prefix is provided, it is prepended to the action execution ID with a hyphen and
+   * together used as the state machine execution name.
+   *
+   * @default - action execution ID
+   */
+  readonly executionNamePrefix?: string;
+}
+
+/**
+ * StepFunctionInvokeAction that is provided by an AWS CodePipeline.
+ */
+export class StepFunctionInvokeAction extends Action {
+  private readonly props: StepFunctionsInvokeActionProps;
+
+  constructor(props: StepFunctionsInvokeActionProps) {
+    super({
+      ...props,
+      resource: props.stateMachine,
+      category: codepipeline.ActionCategory.INVOKE,
+      provider: 'StepFunctions',
+      artifactBounds: {
+        minInputs: 0,
+        maxInputs: 1,
+        minOutputs: 0,
+        maxOutputs: 1,
+      },
+      inputs: (props.stateMachineInput && props.stateMachineInput.inputArtifact) ? [props.stateMachineInput.inputArtifact] : [],
+      outputs: (props.output) ? [props.output] : [],
+    });
+    this.props = props;
+  }
+
+  protected bound(_scope: cdk.Construct, _stage: codepipeline.IStage, options: codepipeline.ActionBindOptions):
+  codepipeline.ActionConfig {
+    // allow pipeline to invoke this step function
+    options.role.addToPolicy(new iam.PolicyStatement({
+      actions: ['states:StartExecution', 'states:DescribeStateMachine'],
+      resources: [this.props.stateMachine.stateMachineArn],
+    }));
+
+    // allow state machine executions to be inspected
+    options.role.addToPolicy(new iam.PolicyStatement({
+      resources: [cdk.Stack.of(this.props.stateMachine).formatArn({
+        service: 'states',
+        resource: 'execution',
+        resourceName: `${this.props.stateMachine.stateMachineArn}:${this.props.executionNamePrefix ?? ''}*`,
+        sep: ':',
+      })],
+    }));
+
+    // allow the Role access to the Bucket, if there are any inputs/outputs
+    if ((this.actionProperties.inputs ?? []).length > 0) {
+      options.bucket.grantRead(options.role);
+    }
+    if ((this.actionProperties.outputs ?? []).length > 0) {
+      options.bucket.grantWrite(options.role);
+    }
+
+    return {
+      configuration: {
+        StateMachineArn: this.props.stateMachine.stateMachineArn,
+        Input: (this.props.stateMachineInput) ? this.props.stateMachineInput.input : [],
+        InputType: (this.props.stateMachineInput) ? this.props.stateMachineInput.inputType : undefined,
```

Comment:
Thanks to TypeScript's safe navigation operator, this can be simplified to:

```suggestion
        InputType: this.props.stateMachineInput?.inputType,
```